src/reducers/game.ts

```diff
@@ -0,0 +1,145 @@
+import { includes } from 'lodash';
+import {newSquare} from '../actions/game';
+import {DOWN, LEFT, MOVE, RIGHT, START, UP} from '../constants';
+
+interface IState {
+  allMoves: string[];
+  board: number[];
+  direction: string;
+  gameOver: boolean;
+  score: number;
+}
+
+const initialState: IState = {
+  allMoves: [],
+  board: [],
+  direction: '',
+  gameOver: false,
+  score: 0,
+};
+
+const isOver = (game: any): boolean => {
+  return false; // TODO
+};
+
+const compressLine = (line: number[]): [boolean, number[], number] => {
+  let points = 0;
+  let size = line.length;
+  let merged: number[] = [];
+  let changed = false;
+  for (let index = 0; index < size; index++) {
+    if (line[index] !== 0) {
+      let newIndex = index;
+      while (newIndex - 1 >= 0 && line[newIndex - 1] === 0) {
+        newIndex -= 1;
+      }
+      if (!includes(merged, newIndex - 1)
+        && line[newIndex - 1] === line[index]) {
+        line[newIndex - 1] *= 2;
+        line[index] = 0;
+        merged.splice(0, 0, newIndex - 1);
+        points += line[newIndex - 1];
+        changed = true;
+      } else if (newIndex !== index) {
+        line[newIndex] = line[index];
+        line[index] = 0;
+        changed = true;
+      }
+    }
+  }
+  return [changed, line, points];
```

Comment:
return Object and make interface out of it.

signature of that function would be:
`const compressLine = (line: ILine): ICompresedLine => {`

where:
```
interface ICompresedLine {
  changed: boolean;
  line:  ILine;
  points: number;
}
```